app/controllers/users_controller.rb

```diff
@@ -11,7 +11,7 @@ def edit
   end
 
   def update
-    require_authentication and return
+    require_authentication or return
```

Comment:
If we are switching to raising errors when authentication fails, the return statement shouldn't be needed. So you should be able to drop the return statement